src/redux/features/asyncUtils.ts

```diff
@@ -7,17 +7,32 @@ export type StatusT<T> = { status: Statuses; data?: T };
 export type Status = StatusT<unknown>; // when data is not relevant in the response
 
 export const buildReducers =
-  <T>(asyncThunk: AsyncThunk<any, any, any>, useArgAsKey?: boolean) =>
+  <T>(asyncThunk: AsyncThunk<any, any, any>, useArgAsKey?: boolean, compositeKeyFn?: (args: unknown) => string) =>
   (builder: CaseReducers<any, any> | ActionReducerMapBuilder<any>) =>
     (builder as any)
-      .addCase(asyncThunk.pending, setStatus<T>('pending', useArgAsKey))
-      .addCase(asyncThunk.fulfilled, setStatus<T>('success', useArgAsKey))
-      .addCase(asyncThunk.rejected, setStatus<T>('error', useArgAsKey));
+      .addCase(asyncThunk.pending, setStatus<T>('pending', useArgAsKey, compositeKeyFn))
+      .addCase(asyncThunk.fulfilled, setStatus<T>('success', useArgAsKey, compositeKeyFn))
+      .addCase(asyncThunk.rejected, setStatus<T>('error', useArgAsKey, compositeKeyFn));
 
 export const setStatus =
-  <T>(status: Statuses, useArgAsKey?: boolean) =>
+  <T>(status: Statuses, useArgAsKey?: boolean, compositeKeyFn?: (arg: unknown) => string) =>
   (state: any, action: any) => {
-    const key = useArgAsKey ? action.meta.arg : action.meta.requestId;
     const data: T = action.payload as T;
+
+    state[action.meta.requestId] = { status, data };
```

Comment:
move this into line 24 condition

Reply:
This was intentional, it allows for accessing by request ID or arg key / composite key. If you still want this change I will need to additionally refactor the places where the request is being fetched by request ID. Let me know, I'm OK either way.

Reply:
Fair enough, this is a good enhancement.